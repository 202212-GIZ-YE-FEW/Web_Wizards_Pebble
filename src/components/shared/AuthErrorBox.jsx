export default function AuthErrorBox(props) {
    const { errors, t } = props;
    return (
        <ul className='bg-warning-600 p-7 mb-4 rounded-lg text-white font-medium'>
            {errors.map((error, index) => {
                return <li key={index}>{t(error)}</li>;
            })}
        </ul>
    );
}
