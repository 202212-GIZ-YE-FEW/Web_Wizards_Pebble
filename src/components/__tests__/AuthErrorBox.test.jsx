import renderer from "react-test-renderer";

import AuthErrorBox from "@/components/shared/AuthErrorBox";

it("renders correctly", () => {
    const tree = renderer
        .create(<AuthErrorBox t={(str) => str} errors={["auth/unknown"]} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
