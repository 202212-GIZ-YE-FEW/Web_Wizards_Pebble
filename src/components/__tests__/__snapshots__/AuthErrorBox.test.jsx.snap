// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="alert__AlertWrapper-sc-4qwf0l-0 iGVtYN alert style-orange-light"
  role="alert"
>
  <button
    aria-label="Close the error alert."
    className="alert-close-button__AlertCloseButtonWrapper-sc-unvdea-0 close-button style-orange-light"
    onClick={[Function]}
  />
  auth/unknown
</div>
`;
